services/src/main/java/org/fao/geonet/api/records/backup/ArchiveAllMetadataJob.java

```diff
@@ -0,0 +1,148 @@
+    private AtomicBoolean backupIsRunning = new AtomicBoolean(false);
+
+
+    @Override
+    protected void executeInternal(JobExecutionContext jobContext) throws JobExecutionException {
+        ServiceContext serviceContext = serviceManager.createServiceContext("unpublishMetadata", context);
+        serviceContext.setLanguage("eng");
+        serviceContext.setAsThreadLocal();
+
+        ApplicationContextHolder.set(this.context);
+
+        if(!settingManager.getValueAsBool(Settings.METADATA_BACKUPARCHIVE_ENABLE)) {
+            Log.info(BACKUP_LOG, "Backup archive not enabled");
+            return;
+        }
+
+        try {
+            createBackup(serviceContext);
+        } catch (Exception e) {
+            Log.error(Geonet.GEONETWORK, "Error running " + ArchiveAllMetadataJob.class.getSimpleName(), e);
+        }
+    }
+
+    public void createBackup(ServiceContext serviceContext) throws Exception {
+
+        ApplicationContext appContext = ApplicationContextHolder.get();
+        GeonetworkDataDirectory dataDirectory = appContext.getBean(GeonetworkDataDirectory.class);
+
+        Path stylePath = dataDirectory.getWebappDir().resolve(Geonet.Path.SCHEMAS);
+
+        if (!backupIsRunning.compareAndSet(false, true)) {
+            return;
+        }
+        long startTime = System.currentTimeMillis();
+        try {
+            Log.info(BACKUP_LOG, "Starting backup of all metadata");
+            System.out.println("Starting backup of all metadata");
+
+            final MetadataRepository metadataRepository = serviceContext.getBean(MetadataRepository.class);
+
+            loginAsAdmin(serviceContext);
+            final Specification<Metadata> harvested = Specifications.where(MetadataSpecs.isHarvested(false)).
+                    and(Specifications.not(MetadataSpecs.hasType(MetadataType.SUB_TEMPLATE)));
+            List<String> uuids = Lists.transform(metadataRepository.findAll(harvested), new Function<Metadata,
+                    String>() {
+                @Nullable
+                @Override
+                public String apply(@Nullable Metadata input) {
+                    return input.getUuid();
+                }
+            });
+
+            Log.info(BACKUP_LOG, "Backing up " + uuids.size() + " metadata");
+
+            String format = "full";
+            boolean resolveXlink = true;
+            boolean removeXlinkAttribute = false;
+            boolean skipOnError = true;
+            Path srcFile = MEFLib.doMEF2Export(serviceContext, new HashSet<>(uuids), format, false, stylePath,
+                    resolveXlink, removeXlinkAttribute, skipOnError);
+
+            Path backupDir = dataDirectory.getBackupDir().resolve(BACKUP_DIR);
+            String today = new SimpleDateFormat("-yyyy-MM-dd").format(new Date());
+            Path destFile = backupDir.resolve(BACKUP_FILENAME + today + ".zip");
+            IO.deleteFileOrDirectory(backupDir);
```

Comment:
If we drop the dir this is not an archive anymore (name is "backup_archive"). Do we expect to keep some kind of history here or is this the responsibility to the end user?  An option would be to add a DELETE operation in the API to clean the archive ?